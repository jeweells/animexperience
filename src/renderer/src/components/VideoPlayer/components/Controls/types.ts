export type ControlsContext = {
  video: HTMLVideoElement | null
}
