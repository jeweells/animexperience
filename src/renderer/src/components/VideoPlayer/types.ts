import { RecentAnimeData } from '../../hooks/useRecentAnimes'
import { Optional } from '@shared/types'
import { VideoOption } from './index'

export type $IframeContents = JQuery<HTMLIFrameElement | Text | Comment | Document>
export type BasicVideoInfo = {
  option?: Optional<VideoOption>
  anime?: Optional<RecentAnimeData>
}

export type KnownOption =
  | 'mega'
  | 'fembed'
  | 'okru'
  | 'mixdrop'
  | 'streamtape'
  | 'stape'
  | 'maru'
  | 'sw'
  | 'yourupload'
