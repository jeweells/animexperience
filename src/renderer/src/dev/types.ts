import { RawDevMessage } from '@shared/types'
import { Moment } from 'moment'

export type DevMessage = RawDevMessage & {
  id: string
  at: Moment
}
