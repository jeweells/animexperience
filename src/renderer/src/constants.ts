export const PUBLIC_URL = 'https://animexp.fun'
export const ONE_HOUR_IN_SECONDS = 60 * 60
export const FAST_SEEK_FORWARD_IN_SECONDS = 5
export const FAST_SEEK_BACKWARD_IN_SECONDS = 6
