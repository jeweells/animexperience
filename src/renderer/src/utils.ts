export const range = (length: number): number[] => {
  return Array(length)
    .fill(0)
    .map((_, x) => x)
}

export const pixel = (n: string | number): string => {
  if (typeof n === 'number') {
    return n + 'px'
  }
  return n
}

export const rendererInvoke = (
  name: keyof typeof window.invokeNames,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ...args: any[]
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
): Promise<any> => {
  return window.electron.ipcRenderer.invoke(window.invokeNames[name].name, ...args)
}

export const random = () => Math.random()
