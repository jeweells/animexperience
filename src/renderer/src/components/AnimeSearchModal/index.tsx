import * as React from 'react'
import { animeSearch } from '@reducers'
import { useAppDispatch, useAppSelector } from '~/redux/utils'
import { TopView } from '@shared/types'
import AnimeSearch from '../AnimeSearch'
import FullModal, { FullModalProps } from '../FullModal'

export type AnimeSearchModalProps = Omit<FullModalProps, 'show' | 'children' | 'view'>

export const AnimeSearchModal: React.FC<AnimeSearchModalProps> = React.memo(({ ...rest }) => {
  const searching = useAppSelector((d) => d.animeSearch.searching)
  const dispatch = useAppDispatch()
  const handleClose = () => {
    dispatch(animeSearch.setSearching(false))
  }
  return (
    <React.Fragment>
      <FullModal
        view={TopView.SEARCH}
        show={searching}
        contrast={true}
        {...rest}
        onPopRequested={handleClose}
      >
        <AnimeSearch onClose={handleClose} />
      </FullModal>
    </React.Fragment>
  )
})

AnimeSearchModal.displayName = 'AnimeSearchModal'

export default AnimeSearchModal
