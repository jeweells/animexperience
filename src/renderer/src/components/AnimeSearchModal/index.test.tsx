import { ThemeProvider } from '@mui/material/styles'
import { Provider } from 'react-redux'
import AnimeSearchModal, { AnimeSearchModalProps } from './index'
import { RootState } from '~/redux/state'
import { DeepPartial } from 'redux'
import { MockStoreEnhanced } from 'redux-mock-store'
import { range } from '~/src/utils'
import { TopView } from '@shared/types'
import theme from '../../theme'

describe('AnimeSearchModal', () => {
  let initialState: DeepPartial<RootState>
  let props: AnimeSearchModalProps
  let store: MockStoreEnhanced
  const onClose = jest.fn()

  const getComponent = (override: Partial<AnimeSearchModalProps> = {}) => {
    store = mockStore(initialState)
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <AnimeSearchModal {...props} {...override} />
        </ThemeProvider>
      </Provider>
    )
  }

  beforeEach(() => {
    initialState = {
      topView: {
        views: [TopView.SEARCH]
      },
      animeSearch: {
        status: {
          result: 'succeeded',
          moreResults: 'succeeded'
        },
        result: {
          hasNext: true,
          search: 'anime name to search',
          matches: range(12).map((n) => ({
            image: 'image url ' + n,
            link: 'anime link ' + n,
            name: 'anime name ' + n
          })),
          maxPage: 5,
          nextPage: 2
        }
      }
    }

    props = {
      onClose
    }
    onClose.mockReset()
  })

  it('renders default', () => {
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('should not render when is not the corresponding top view', () => {
    initialState.topView!.views = []
    const wrapper = render(getComponent())
    // asFragment() won't work since the modal is outside the react root element
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('closing it should pop topView', () => {
    render(getComponent())
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        type: 'topView/pop'
      })
    )
  })
})
