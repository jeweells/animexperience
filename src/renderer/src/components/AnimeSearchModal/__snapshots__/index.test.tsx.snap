// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeSearchModal renders default 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  margin: 0;
  width: 100vw;
  overflow: hidden;
  --modal-height: calc(100vh - 56px);
  height: var(--modal-height);
  background-color: transparent;
  top: 56px!important;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-0 .MuiDialog-container {
  background-color: #0f131a;
}

.emotion-0 .MuiPaper-root {
  background-image: unset;
  background-color: #0f131a;
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #121212;
  color: #e9ebf0;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  background-image: linear-gradient(rgba(255, 255, 255, 0.16), rgba(255, 255, 255, 0.16));
  margin: 0;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: none;
  max-width: 100%;
  width: 100%;
  height: 100%;
  border-radius: 0;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

.emotion-3.MuiDialog-paperScrollBody {
  margin: 0;
  max-width: 100%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  overflow-y: auto;
  padding-bottom: 32px;
  height: 100%;
  position: relative;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  padding: 32px 64px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
}

.emotion-7 {
  font-weight: 500;
  color: #e7e7e7;
  font-size: 1.2rem;
  line-height: 1;
  text-shadow: 0 0 2px black,0 0 6px black;
  margin-bottom: 24px;
}

.emotion-8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #fff;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 36px;
  min-width: 36px;
  color: #e9ebf0;
  padding: 8px 6px;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 1.2rem;
  line-height: 1.42857143;
  border-radius: 6px;
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9:hover {
  background-color: rgba(255, 255, 255, 0.08);
}

@media (hover: none) {
  .emotion-9:hover {
    background-color: transparent;
  }
}

.emotion-9.Mui-disabled {
  background-color: transparent;
  color: rgba(255, 255, 255, 0.3);
}

.emotion-9:hover {
  background-color: #3c3f43;
}

.emotion-9.Mui-disabled {
  color: rgb(255 255 255 / 25%);
}

.emotion-10 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.emotion-11 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  row-gap: 24px;
  -webkit-column-gap: 4px;
  column-gap: 4px;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
}

.emotion-13 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width: 7396px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 22 * 16px - 2 *
                72px
            ) / 23
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 7080px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 21 * 16px - 2 *
                72px
            ) / 22
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 6764px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 20 * 16px - 2 *
                72px
            ) / 21
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 6448px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 19 * 16px - 2 *
                72px
            ) / 20
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 6132px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 18 * 16px - 2 *
                72px
            ) / 19
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 5816px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 17 * 16px - 2 *
                72px
            ) / 18
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 5500px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 16 * 16px - 2 *
                72px
            ) / 17
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 5184px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 15 * 16px - 2 *
                72px
            ) / 16
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 4868px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 14 * 16px - 2 *
                72px
            ) / 15
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 4552px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 13 * 16px - 2 *
                72px
            ) / 14
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 4236px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 12 * 16px - 2 *
                72px
            ) / 13
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 3920px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 11 * 16px - 2 *
                72px
            ) / 12
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 3604px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 10 * 16px - 2 *
                72px
            ) / 11
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 3288px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 9 * 16px - 2 *
                72px
            ) / 10
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 2972px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 8 * 16px - 2 *
                72px
            ) / 9
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 2656px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 7 * 16px - 2 *
                72px
            ) / 8
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 2340px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 6 * 16px - 2 *
                72px
            ) / 7
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 2024px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 5 * 16px - 2 *
                72px
            ) / 6
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 1708px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 4 * 16px - 2 *
                72px
            ) / 5
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 1392px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 3 * 16px - 2 *
                72px
            ) / 4
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 1076px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 2 * 16px - 2 *
                72px
            ) / 3
        );
    --height: calc(var(--width) * 0.94);
  }
}

@media (max-width: 760px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 1 * 16px - 2 *
                72px
            ) / 2
        );
    --height: calc(var(--width) * 0.94);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width: 444px) {
  .emotion-13 {
    --width: calc(
          (
              100vw - 0 * 16px - 2 *
                72px
            ) / 1
        );
    --height: calc(var(--width) * 0.94);
  }
}

.emotion-13 .ManageButtons {
  opacity: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.emotion-13:hover {
  z-index: 10;
}

.emotion-13:hover .ManageButtons {
  opacity: 1;
}

.emotion-14 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.emotion-15 {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  background-color: #292d33;
  padding: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-17 {
  font-weight: 400;
  color: white;
  font-size: 1rem;
  line-height: 1;
  margin-bottom: 4px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="full-modal"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen emotion-3"
        role="dialog"
      >
        <div
          class="emotion-4"
          style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="emotion-5"
            style="padding: 32px 72px;"
          >
            <div
              class="emotion-6"
              style="align-items: center;"
            >
              <h2
                class="emotion-7"
                style="font-size: 1.5rem; white-space: pre;"
              >
                Resultados de 
                <span
                  style="opacity: 0.8;"
                >
                  "
                  anime name to search
                  "
                </span>
              </h2>
              <div
                class="emotion-8"
              />
              <button
                aria-label="Cerrar"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-9"
                data-mui-internal-clone-element="true"
                data-testid="anime-search-close-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-10"
                  data-testid="CloseRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.3 5.71a.9959.9959 0 0 0-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root emotion-11"
                />
              </button>
            </div>
            <div
              class="emotion-12"
              style="column-gap: 16px;"
            >
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 0"
                        class="emotion-15"
                        src="image url 0"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 0"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 0
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 1"
                        class="emotion-15"
                        src="image url 1"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 1"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 1
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 2000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 2000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 2"
                        class="emotion-15"
                        src="image url 2"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 2"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 2
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 2500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 2500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 3"
                        class="emotion-15"
                        src="image url 3"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 3"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 3
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 3000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 3000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 4"
                        class="emotion-15"
                        src="image url 4"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 4"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 4
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 3500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 3500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 5"
                        class="emotion-15"
                        src="image url 5"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 5"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 5
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 4000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 4000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 6"
                        class="emotion-15"
                        src="image url 6"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 6"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 6
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 4500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 4500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 7"
                        class="emotion-15"
                        src="image url 7"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 7"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 7
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 5000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 5000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 8"
                        class="emotion-15"
                        src="image url 8"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 8"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 8
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 5500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 5500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 9"
                        class="emotion-15"
                        src="image url 9"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 9"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 9
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 6000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 6000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 10"
                        class="emotion-15"
                        src="image url 10"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 10"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 10
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  data-testid="anime-entry-wrapper"
                  style="opacity: 1; webkit-transition: opacity 6000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 6000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="emotion-13"
                    data-testid="anime-entry-background"
                  >
                    <div
                      class="emotion-14"
                    >
                      <img
                        alt="anime name 11"
                        class="emotion-15"
                        src="image url 11"
                      />
                    </div>
                    <div
                      class="emotion-16 AnimeInfo"
                      data-testid="anime-details-entry-anime-info"
                    >
                      <h3
                        aria-label="anime name 11"
                        class=" emotion-17"
                        data-mui-internal-clone-element="true"
                      >
                        anime name 11
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-testid="waypoint-div"
              >
                [Waypoint]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AnimeSearchModal should not render when is not the corresponding top view 1`] = `
<body
  style=""
>
  <div />
</body>
`;
